### ISSUE TITLE
Move no-implicit-explicit-label check into label-content-name-mismatch rule

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#4070]
Part of https://github.com/dequelabs/axe-core/issues/4066

~~The `no-implicit-explicit-label` check from `aria-input/toggle-field-name` will be moved to the `label-content-name-mismatch` experimental rule, and we'll downscale the impact from `serious` to `moderate`.~~

## Update

After taking a closer look I no longer think the above was right. `no-implicit-explicit-label` never fails. It says that code like the following needs review:

```html
<label>
  first name
  <div role="textbox" aria-label="name"></div>
</label>
```

This actually seems like a serious issue to me, if not critical. I think leaving this at moderate is incorrect. So instead of moving this check to an experimental rule, bumping the impact from `moderate` to `serious` is appropriate. User impact will be minimal as needs-review issues aren't counted in impact numbers.

**Comment by padmavemulapati** [comment]
Validated using latest axe-core develop branch code base,
for the code:

```
<label>
  first name
  <div role="textbox" aria-label="name"></div>
</label>
```
seeing the issue in `incomeplete` and `serious`



![Image](https://github.com/dequelabs/axe-core/assets/47409579/f8c3aae3-0285-45ef-8e45-d60293aeb6a7)
